src/Traits/Shipments.php

```
@@ -58,7 +58,14 @@ public function createShipment(
             $params,
         );
 
-        if (! $response['has_errors'] && $config->asObject) {
+        if ($this->responseIsRateLimit($response)) {
+            throw new Exception('API rate limit exceeded');
```

I wasn't sure if we should throw an exception or create an error entry in the response. According to how I've seen this used it felt like throwing an exception would be the best fit, I'm open to suggestions though